cpp/src/groupby/sort/group_collect.cu

```diff
@@ -85,24 +85,26 @@ std::unique_ptr<column> group_collect(column_view const &values,
                                       rmm::cuda_stream_view stream,
                                       rmm::mr::device_memory_resource *mr)
 {
-  auto offsets_column = make_numeric_column(
-    data_type(type_to_id<size_type>()), num_groups + 1, mask_state::UNALLOCATED, stream, mr);
-
-  thrust::copy(rmm::exec_policy(stream),
-               group_offsets.begin(),
-               group_offsets.end(),
-               offsets_column->mutable_view().template begin<size_type>());
-
-  std::unique_ptr<cudf::column> child_column;
-
-  // If column of grouped values contains null elements, and null_policy == EXCLUDE,
-  // those elements must be filtered out, and offsets recomputed.
-  if (null_handling == null_policy::EXCLUDE && values.has_nulls()) {
-    std::tie(child_column, offsets_column) = cudf::groupby::detail::purge_null_entries(
-      values, offsets_column->view(), num_groups, stream, mr);
-  } else {
-    child_column = std::make_unique<cudf::column>(values, stream, mr);
-  }
+  auto [child_column,
+        offsets_column] = [null_handling, num_groups, &values, &group_offsets, stream, mr] {
+    auto offsets_column = make_numeric_column(
+      data_type(type_to_id<size_type>()), num_groups + 1, mask_state::UNALLOCATED, stream, mr);
+
+    thrust::copy(rmm::exec_policy(stream),
+                 group_offsets.begin(),
+                 group_offsets.end(),
+                 offsets_column->mutable_view().template begin<size_type>());
+
+    // If column of grouped values contains null elements, and null_policy == EXCLUDE,
+    // those elements must be filtered out, and offsets recomputed.
+    if (null_handling == null_policy::EXCLUDE && values.has_nulls()) {
+      return cudf::groupby::detail::purge_null_entries(
```

Comment:
Not there yet. You generated the `offsets_column` column, then fill it with `thrust::copy`, then you return something else!